themes/base.theme.bash

```diff
@@ -83,10 +83,19 @@ RBENV_THEME_PROMPT_SUFFIX='|'
 RBFU_THEME_PROMPT_PREFIX=' |'
 RBFU_THEME_PROMPT_SUFFIX='|'
 
-GIT_EXE=`which git 2> /dev/null || true`
-P4_EXE=`which p4 2> /dev/null || true`
-HG_EXE=`which hg  2> /dev/null || true`
-SVN_EXE=`which svn  2> /dev/null || true`
+GIT_EXE=$(which git 2> /dev/null || true)
+P4_EXE=$(which p4 2> /dev/null || true)
+HG_EXE=$(which hg  2> /dev/null || true)
+SVN_EXE=$(which svn  2> /dev/null || true)
+
+# Check for broken SVN exe that is caused by some versions of Xcode.
+# See https://github.com/Bash-it/bash-it/issues/1612 for more details.
+if [[ -x "$SVN_EXE" ]] ; then
+  if "$SVN_EXE" 2>&1 | grep -q "subversion command line tools are no longer provided" ; then
```

Comment:
This is oddly specific and doesn't include a check for OSX ...

 Maybe we could try something like:
```
"$SVN_EXE" --version >/dev/null 2>&1 || export SVN_EXE=""
```

Assuming that the stub returns an error code for  `--version` (I'm not able to test locally)

Reply:
Good point - I thought about that as well. I don't have the broken `svn` command, since I downloaded the Xcode command line tools and my `svn` command works just fine.

Maybe someone with a broken `svn` binary can comment on the output and return code of `svn --version`? @arturmartins?

Reply:
> It's exactly the same with or without `--version`.

@davesteinberg awesome,

The the purpose of informing what the final PR here might look like, can you confirm that the following command:
```
$ svn --version >/dev/null 2>&1 || echo "svn --version failed"
```

Generates this output:
```
svn --version failed
```

Thanks again!

-DF

Reply:
@davidpfarrell Yup, it does.

Reply:
Thanks for the help - I've adjusted the code to check for the return code instead of the error message.

Ideally, we should have a unit case for this as well, I'll try to think of a way to mock the `svn` command for the individual cases.